### PR TITLE
[Legacy] Adds google analytics to netmanager

### PR DESCRIPTION
**Pull request by Codebmk**
#### Summary of Changes (What does this PR do?)
- Adds google analytics to netmanager

#### Is this change ready to hit production in its current state?
- [x] Yes
- [ ] No

#### Status of maturity (all need to be checked before merging):

- [x] I consider this code done
- [ ] I've tested this locally
- [ ] This branch has been LGTM'ed by a reviewer

#### How should this be manually tested?
* Please include the steps to be done inorder to setup and test this PR.

#### What are the relevant tickets?
- [AN-220](https://airqoteam.atlassian.net/browse/AN-220?atlOrigin=eyJpIjoiY2MyNTIxMzM4Y2ZhNGE1ZGE2YzA4MTA2MWE1ZWYyNzkiLCJwIjoiaiJ9)

#### Screenshots (optional)

### CODE REVIEW

File: netmanager/public/index.html [hunk 1]
```diff
@@ -18,43 +20,75 @@
...
+      src="https://www.googletagmanager.com/gtag/js?id=%REACT_APP_LEGACY_GOOGLE_ANALYTICS_TOKEN%"
+    ></script>
+    <script>
+      window.dataLayer = window.dataLayer || [];
+      function gtag() {
+        dataLayer.push(arguments);
+      }
+      gtag('js', new Date());
+
+      gtag('config', '%REACT_APP_LEGACY_GOOGLE_ANALYTICS_TOKEN%');
```
Comment (123MwanjeMike) [inline netmanager/public/index.html:77]: Hi @Codebmk 
Could we also add this variable to the [.env.sample](https://github.com/airqo-platform/AirQo-frontend/blob/3d4ee46d2793ad1d34124ece4e409f7515d05ff7/netmanager/.env.sample) file with a dummy value?

**Review by 123MwanjeMike — CHANGES_REQUESTED** [review]
Thank you @Codebmk 
One request tho...

File: netmanager/public/index.html (hunk 1 above)
Comment (123MwanjeMike) [reply netmanager/public/index.html:77]: Also if possible, can we rename the variable to `REACT_APP_GOOGLE_ANALYTICS_TOKEN` as an abstraction mechanism. The codebase should not know that it's even legacy.

**Review by 123MwanjeMike — APPROVED** [review]
Thanks @Codebmk 
LGTM

**Comment by Codebmk** [comment]
Works for deploy preview
deploy analytics

File: .github/workflows/deploy-frontend-pr-previews.yml [hunk 2]
```diff
@@ -79,9 +79,6 @@ jobs:
     needs: [check, branch-name]
     if: needs.check.outputs.run_platform == 'true'
     runs-on: ubuntu-latest
-    permissions:
```
[the remark this answers is not in the document]
Comment (Codebmk) [reply .github/workflows/deploy-frontend-pr-previews.yml:82]: Whoops help remove those @123MwanjeMike

[the remark this answers is not in the document]
Comment (123MwanjeMike) [reply .github/workflows/deploy-frontend-pr-previews.yml:82]: That's a review comment.
They were removed

**Review by 123MwanjeMike — APPROVED** [review]

**Comment by 123MwanjeMike** [comment]
image

@Codebmk , please note that this new change means that users with ad blockers will get white screen, until ad blockers are removed. 

Otherwise, great work

**Comment by Codebmk** [comment]
> image
> 
> @Codebmk , please note that this new change means that users with ad blockers will get white screen, until ad blockers are removed.
> 
> Otherwise, great work

@123MwanjeMike I don't think so. Is it because it is a deploy preview?

**Review by Baalmart — APPROVED** [review]
LGTM

**Comment by Baalmart** [comment]
Hi @Codebmk , could you add the status of this PR? Is it a `read-to-review`? cc: @123MwanjeMike

File: netmanager/.env.sample [hunk 3]
```diff
@@ -13,4 +13,5 @@ REACT_APP_CLOUDINARY_PRESET=abcdef
 REACT_APP_MAPBOX_TOKEN=pk.ey123abc
 REACT_APP_JIRA_KEY=abcdef
 NODE_PATH=src/
-PORT=5000
\ No newline at end of file
+PORT=5000
+REACT_APP_GOOGLE_ANALYTICS_TOKEN='G-HJTULVGQ8W'
```
Comment (123MwanjeMike) [inline netmanager/.env.sample:17]: Can we remove the enclosing quotes @Codebmk?

**Review by 123MwanjeMike — CHANGES_REQUESTED** [review]

File: netmanager/.env.sample (hunk 3 above)
Comment (Codebmk) [reply netmanager/.env.sample:17]: Those have no effect honestly. Thanks.

**Comment by 123MwanjeMike** [comment]
> @123MwanjeMike I don't think so. Is it because it is a deploy preview?

Hi @Codebmk , I know for a fact that what we see in a deploy preview is what would happen in the staging/prod environments. That kind of behaviour is normal and expected given that the token tracks users activity. You just have to be aware of the fact that ad blockers are made to block these trackers.

The comment was meant to make you aware of this and perhaps prompt you to put in place a mechanism to ask users to turn off ad blockers for the platform for a smooth interaction.

File: netmanager/.env.sample (hunk 3 above)
Comment (123MwanjeMike) [reply netmanager/.env.sample:17]: Is there a trimming mechanism to get rid of them in the codebase?

Comment (123MwanjeMike) [reply netmanager/.env.sample:17]: Look at the .env.sample as a documentational file. It helps new users setup their env values basing on what they see. That's why its best to have the (dummy)value exactly as is supposed to be.

**Review by 123MwanjeMike — APPROVED** [review]

**Pull request merged**